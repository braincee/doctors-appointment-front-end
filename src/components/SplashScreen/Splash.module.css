.background {
  position: relative;
  min-height: 100vh;
}

.background::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background-image: url('../assets/splashbackground.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: 50%;
}

.background::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: #f7ba01b9;
  opacity: 0.7;
}

.header {
  z-index: 3;
  font-size: 75px;
  letter-spacing: 1px;
  margin: 80px 0 100px;
}

.btncontainer {
  z-index: 1;
  margin: 80px 0 200px;
}

@media screen and (max-width: 768px) {
  .splashbackground {
    margin-left: 0;
  }

  .header {
    font-size: 50px;
  }
}
