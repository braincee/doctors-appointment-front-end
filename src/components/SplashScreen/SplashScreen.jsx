import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Splash.module.css';

const SplashScreen = () => (
  <div className={`${styles.background} container-fluid d-flex flex-column align-items-center justify-content-around vw-100`}>
    <h1 className={`${styles.header} align-items-center`}>Doctors Appointment</h1>
    <div className={`${styles.btncontainer}`}>
      <NavLink to="login">
        <button type="button" className="btn btn-warning me-5 fs-5">Log in</button>
      </NavLink>
      <NavLink to="signup">
        <button type="button" className="btn btn-info me-5 fs-5">Sign up</button>
      </NavLink>
    </div>
  </div>
);

export default SplashScreen;
