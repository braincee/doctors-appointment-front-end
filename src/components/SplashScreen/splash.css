.background {
  position: relative;
  min-height: 100vh;
}

.background::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  background-image: url('../../assets/splashbackground.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: 50%;
}

.background::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: #64636062;
  opacity: 0.7;
}

.header {
  z-index: 3;
  font-size: 75px;
  letter-spacing: 1px;
  margin: 80px 0 100px;
}

.btncontainer {
  z-index: 1;
  margin: 80px 0 200px;
}

.button1:hover {
  color: #000000e5;
  background-color: #555453b4;
  font-family: 'Times New Roman', Times, serif;
}

@media screen and (max-width: 768px) {
  .splashbackground {
    margin-left: 0;
  }

  .header {
    font-size: 50px;
  }
}

@media only screen and (max-width: 540px) {
  .header {
    position: relative;
    font-size: 20px;
    left: -100px;
    color: aliceblue;
  }
}
