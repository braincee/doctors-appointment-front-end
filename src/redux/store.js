import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';




const initialState = {};
const middleware = [thunk];

const store = createStore(initialState, (applyMiddleware(...middleware)));