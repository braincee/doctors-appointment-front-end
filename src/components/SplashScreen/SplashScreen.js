import React from 'react';
import { NavLink } from 'react-router-dom';
import './splash.css';

const SplashScreen = () => (
  <div className="container-fluid d-flex flex-column align-items-center justify-content-around vw-100 background">
    <h1 className="header align-items-center">Doctors Appointment</h1>
    <div className="btncontainer">
      <NavLink to="/login">
        <button type="button" className="btn btn-secondary me-5 fs-5 button1">Log in</button>
      </NavLink>
      <NavLink to="/signup">
        <button type="button" className="btn btn-secondary me-5 fs-5 button1">Sign up</button>
      </NavLink>
    </div>
  </div>
);

export default SplashScreen;
