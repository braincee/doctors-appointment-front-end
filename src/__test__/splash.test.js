import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import SplashScreen from '../components/SplashScreen/SplashScreen';
import store from '../redux/configureStore';

describe('Splash screen tests', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <SplashScreen />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Splash page has a sign up link', () => {
    const register = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <SplashScreen />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(register.findByText('Sign up')).toBeTruthy();
  });

  it('Splash page has a log in link', () => {
    const register = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <SplashScreen />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(register.findByText('Log in')).toBeTruthy();
  });
});
