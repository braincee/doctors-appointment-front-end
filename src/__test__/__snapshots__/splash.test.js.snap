// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash screen tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="background container-fluid d-flex flex-column align-items-center justify-content-around vw-100"
      >
        <h1
          class="header align-items-center"
        >
          Doctors Appointment
        </h1>
        <div
          class="btncontainer"
        >
          <a
            class=""
            href="/login"
          >
            <button
              class="button1 btn btn-secondary me-5 fs-5"
              type="button"
            >
              Log in
            </button>
          </a>
          <a
            class=""
            href="/signup"
          >
            <button
              class="button1 btn btn-secondary me-5 fs-5"
              type="button"
            >
              Sign up
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="background container-fluid d-flex flex-column align-items-center justify-content-around vw-100"
    >
      <h1
        class="header align-items-center"
      >
        Doctors Appointment
      </h1>
      <div
        class="btncontainer"
      >
        <a
          class=""
          href="/login"
        >
          <button
            class="button1 btn btn-secondary me-5 fs-5"
            type="button"
          >
            Log in
          </button>
        </a>
        <a
          class=""
          href="/signup"
        >
          <button
            class="button1 btn btn-secondary me-5 fs-5"
            type="button"
          >
            Sign up
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
